source/PlayerInfo.h

```diff
@@ -307,6 +307,7 @@ class PlayerInfo {
 	bool SelectShips(const Rectangle &box, bool hasShift);
 	bool SelectShips(const std::vector<const Ship *> &stack, bool hasShift);
 	void SelectShip(const Ship *ship, bool hasShift);
+	void DeselectShip(const std::shared_ptr<Ship> ship);
```

Comment:
Since it's only used for comparison, we should probably just pass a `const Ship *`, not a whole `std::shared_ptr<Ship>`.